platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/service/WritableRosterStoreTest.java

```diff
@@ -0,0 +1,208 @@
+
+        rosterStateAccessor = new ReadableRosterStore(writableStates);
+        rosterStateModifier = new WritableRosterStore(writableStates);
+    }
+
+    @Test
+    @DisplayName("Test determine active roster without software upgrade")
+    void testSetCandidateRosterWithValidInputs() {
+        final Roster roster1 = createValidRoster(1);
+        rosterStateModifier.setCandidateRoster(roster1);
+
+        assertEquals(rosterStateAccessor.getCandidateRoster(), roster1);
+
+        final Roster roster2 = createValidRoster(2);
+        rosterStateModifier.setCandidateRoster(roster2);
+        assertEquals(roster2, rosterStateAccessor.getCandidateRoster());
+    }
+
+    @Test
+    void testSetCandidateRosterWhenRosterIsNull() {
+        assertThrows(NullPointerException.class, () -> rosterStateModifier.setCandidateRoster(null));
+    }
+
+    @Test
+    @DisplayName("Tests that adopting a candidate roster throws an exception when the candidate roster is null.")
+    void testAdoptCandidateRosterWhenCandidateRosterNotFound() {
+        enableSoftwareUpgradeMode(true);
+        final Exception exception = assertThrows(
+                NullPointerException.class, () -> rosterStateModifier.determineActiveRoster(version, initialState));
+        assertEquals(
+                "Candidate Roster must be present in the state before attempting to adopt a roster.",
+                exception.getMessage());
+    }
+
+    @Test
+    @DisplayName("Tests that adopting a candidate roster returns the candidate roster as the active roster.")
+    void testAdoptCandidateRosterWithValidCandidateRoster() {
+        enableSoftwareUpgradeMode(true);
+        final Roster candidateRoster = createValidRoster(1);
+        rosterStateModifier.setCandidateRoster(candidateRoster);
+        assertEquals(rosterStateAccessor.getCandidateRoster(), candidateRoster);
+        assertNull(rosterStateAccessor.getActiveRoster());
+
+        rosterStateModifier.determineActiveRoster(version, initialState);
+        assertSame(candidateRoster, rosterStateAccessor.getActiveRoster());
+    }
+
+    @Test
+    @DisplayName(
+            "Test determine active roster during normal restart but with no active roster present in the state throws exception")
+    void testDetermineActiveRosterDuringNormalRestartWithoutActiveRoster() {
+        enableSoftwareUpgradeMode(false);
+        Exception exception = assertThrows(
+                NullPointerException.class, () -> rosterStateModifier.determineActiveRoster(version, initialState));
+        assertEquals(
+                "Active Roster must be present in the state during normal network restart.", exception.getMessage());
+    }
+
+    @Test
+    @DisplayName("Test determine active roster during normal restart but with an active roster present")
+    void testDetermineActiveRosterDuringNormalRestartWithActiveRoster() {
+        enableSoftwareUpgradeMode(true);
```

Comment:
A comment explaining why `enableSoftwareUpgradeMode(true)` is needed would be helpful.